src/app.js

```diff
@@ -207,6 +222,61 @@ gpii.app.exit = function (that) {
     }
 };
 
+/**
+ * Extracts data for the user's preference sets (including the active preference
+ * set) from the message received when the user keys in.
+ * @param message {Object} The message sent when the user keys is (a JSON
+ * object).
+ * @return {Object} An object containing all preference sets and the active preference
+ * set.
+ */
+gpii.app.extractPreferencesData = function (message) {
+    var value = message.value || {},
+        preferences = value.preferences || {},
+        contexts = preferences.contexts,
+        activeContextName = value.activeContextName,
+        sets = [],
+        activeSet = null;
+
+    if (contexts) {
+        sets = fluid.hashToArray(contexts, "path");
+        activeSet = fluid.find_if(sets, function (preferenceSet) {
+            return preferenceSet.path === activeContextName;
+        });
+    }
+
+    return {
+        sets: sets,
+        activeSet: activeSet
+    };
+};
+
+/**
+ * Opens connection to the PCP Channel WebSocket
+ * @param that {Object} The app
+ */
+gpii.app.addCommunicationChannel = function (that) {
+    var socket = new ws("ws://localhost:8081/pcpChannel"); // eslint-disable-line new-cap
+
+    socket.on("message", function (rawData) {
+        var data = JSON.parse(rawData),
+            operation = data.type,
+            path = data.path,
+            preferences;
+
+        if (operation === "ADD") {
+            if (path.length === 0) {
+                preferences = gpii.app.extractPreferencesData(data);
+                that.updatePreferences(preferences);
+                // TODO: Extract settings for the preference set and visualize them.
```

Comment:
Are we going with a model where the electron app is a separate client of the PCPChannel or will the message be relayed here?